packages/dds/merge-tree/src/attributionCollection.ts

```diff
@@ -187,6 +205,49 @@ export class AttributionCollection implements IAttributionCollection<Attribution
 		return this.get(this.findIndex(offset));
 	}
 
+	public getKeysInOffsetRange(
+		startOffset: number,
+		endOffset?: number,
+	): { offset: number; key: AttributionKey }[];
+	public getKeysInOffsetRange(
+		startOffset: number,
+		endOffset?: number,
+		channel?: string,
+	): { offset: number; key: AttributionKey }[] | undefined;
+	public getKeysInOffsetRange(
+		startOffset: number,
+		endOffset?: number,
+		channel?: string,
+	): { offset: number; key: AttributionKey }[] | undefined {
+		assert(startOffset >= 0 && startOffset < this._length, "startOffset should be valid");
```

Comment:
users will call this but not system code, so we should use `UsageError` here for this validation.